src/tracker.rs

```diff
@@ -105,17 +105,18 @@ impl Tracker {
     ) -> Result<Option<(BlockHash, Transaction)>> {
         // Note: there are two blocks with coinbase transactions having same txid (see BIP-30)
         let blockhashes = self.index.filter_by_txid(txid);
-        let mut result = None;
-        daemon.for_blocks(blockhashes, |blockhash, block| {
-            if result.is_some() {
-                return; // keep first matching transaction
-            }
```

Comment:
This optimization is currently removed by this MR, is BIP-30 the only case or there are others?

Reply:
Indeed, currently there are only two cases of collisions - https://github.com/romanz/electrs/pull/928#issuecomment-1694755943

Reply:
With this MR it's not granted blocks are processed in height order so it's not necessarily the first occurrence of the tx to be returned.
Since there are only 2 txids, would it be ok to hardcode those?

Reply:
currently there are these 2, but bip 30 allows txids to collide with existing ones when no new output points would overwrite existing ones in the utxo set (the reason why bip 30 was made, more on this in the bip document)

for optimization purposes i think you can assume no collisions are there (and hardcode the 2 known ones), if it still outputs _a_ transaction with that txid when there is in fact a collision. just no panic or ub